docs/baseband/tutorials/getting_started.rst

```diff
@@ -0,0 +1,388 @@
+
+    >>> fw.sample_shape
+    SampleShape(nchan=8)
+
+In this case, the required dimensions are the same as the arrays from
+``fr.read``.  We can thus write the data to file using::
+
+    >>> while fr.tell() < fr.size:
+    ...     fw.write(fr.read(fr.samples_per_frame))
+    >>> fr.close()
+    >>> fw.close()
+
+For our sample file, we could simply have written
+
+    ``fw.write(fr.read())``
+
+instead of the loop, but for large files, reading and writing should be done in
+smaller chunks to minimize memory usage.  Baseband stores only the data frame
+or frame set being read or written to in memory.
+
+We can check the validity of our new file by re-opening it::
+
+    >>> fr = vdif.open(SAMPLE_VDIF, 'rs')
+    >>> fh = vdif.open('test_vdif.vdif', 'rs')
+    >>> fh.sample_shape
+    SampleShape(nchan=8)
+    >>> np.all(fr.read() == fh.read())
+    True
+    >>> fr.close()
+    >>> fh.close()
+
+File Format Conversion
+----------------------
+
+It is often preferable to convert data from one file format to another that
+offers wider compatibility, or better fits the structure of the data.  As an
+example, we convert the sample Mark 4 data to VDIF.
+
+Since we don't have a VDIF header handy, we pass the relevant Mark 4 header
+values into `vdif.open <baseband.vdif.open>` to create one. 
+
+    >>> import baseband.mark4 as mark4
+    >>> from baseband.data import SAMPLE_MARK4
+    >>> fr = mark4.open(SAMPLE_MARK4, 'rs', ntrack=64, decade=2010)
+    >>> spf = 640       # fanout * 160 = 640 invalid samples per Mark 4 frame
+    >>> f_rate = (fr.frames_per_second * fr.samples_per_frame / spf)*u.Hz
+    >>> fw = vdif.open('m4convert.vdif', 'ws', edv=1, nthread=1,
+    ...                samples_per_frame=spf, nchan=fr.sample_shape.nchan,
+    ...                framerate=f_rate, complex_data=fr.complex_data, 
+    ...                bps=fr.bps, time=fr.time0)
+
+We choose ``edv = 1`` since it's the simplest VDIF EDV whose header includes a
+frame rate. The concept of threads does not exist in Mark 4, so the file
+effectively has ``nthread = 1``.  As discussed in the :ref:`Mark 4
+documentation <mark4>`, the data at the start of each frame is effectively
+overwritten by the header and are represented by invalid samples in the stream
+reader.  We set ``samples_per_frame`` to ``640`` so that each section of
+invalid data is captured in a single frame.  The framerate is then set to 50
+kHz for consistency.
+
+We now write the data to file, manually flagging each invalid data frame::
+
+    >>> while fr.tell() < fr.size:
+    ...     d = fr.read(fr.samples_per_frame)
+    ...     fw.write(d[:640], invalid_data=True)
+    ...     fw.write(d[640:])
+    >>> fr.close()
+    >>> fw.close()
+
+Lastly, we check our new file::
+
+    >>> fr = mark4.open(SAMPLE_MARK4, 'rs', ntrack=64, decade=2010)
+    >>> fh = vdif.open('m4convert.vdif', 'rs')
+    >>> np.all(fr.read() == fh.read())
+    True
+    >>> fr.close()
+    >>> fh.close()
```

Comment:
Let's add a warning here at the end, that for general conversion, one has to take care to properly scale decoded data to the scale needed by the other format; only for VLBI data formats (VDIF, Mark4, Mark5B) are the scales the same. It may not be strange for now to point to the conversion tests `test_conversion.py`.

Reply:
I've added a short paragraph about this at the end.  Once we have a gallery of conversion examples in the documentation, I'll point readers to that rather than `test_conversion.py`.